chatlas/_chat.py

```diff
@@ -807,6 +810,181 @@ def console(
             self.chat(user_input, echo=echo, stream=stream, kwargs=kwargs)
             print("")
 
+    def to_solver(self):
+        """
+        Return an InspectAI solver that proxies prompts through this chat.
+
+        This method creates an InspectAI solver that uses this Chat instance to
+        generate responses. The solver automatically translates the chat's
+        conversation history (including system prompt, previous turns, and rich
+        content) into InspectAI's message format.
+
+        Returns
+        -------
+        Solver
+            An InspectAI solver function that can be used with InspectAI's
+            evaluation framework.
+        """
+
+        try:
+            import inspect_ai.model as imodel
+            import inspect_ai.solver as isolver
+            import inspect_ai.tool as itool
+        except ImportError as exc:  # pragma: no cover - optional dependency
+            raise ImportError(
+                "Chat.to_solver() requires the optional dependency `inspect-ai`. "
+                "Install it with `pip install inspect-ai`."
+            ) from exc
+
+        def turn_as_messages(turn: Turn) -> list[imodel.ChatMessage]:
```

Comment:
I think it'll probably make sense to pull these inspect helper functions into a separate (inspect focused) module